File: WebApplication/ClientApp/src/app/_orchestration/getLeaderboard/getRaceLeaderboard/raceLeaderboardDto.ts
import { IIntervalType } from "../../../_common/IIntervalName";
import { IRaceSeriesType } from "../../../_common/IRaceSeriesType";
import { LocationInfoWithRank } from "../../locationInfoWithRank";
import { LeaderboardResultDto } from "../leaderboardResultDto";

export interface RaceLeaderboardDto extends IRaceSeriesType {
  raceName: string
  raceSeriesDescription: string
  raceKickOffDate: string
  locationInfoWithRank: LocationInfoWithRank
  leaderboards: RaceLeaderboardByCourseDto[]
}

export interface RaceLeaderboardByCourseDto extends IIntervalType {
  courseId: number
  courseName: string
  sortOrder: number
  highestIntervalName: string
  results: LeaderboardResultDto[]
}

File: Orchestration/GetLeaderboard/GetRaceLeaderboard/RaceLeaderboardDto.cs
namespace Orchestration.GetLeaderboard.GetRaceLeaderboard;

public record RaceLeaderboardDto
{
    public required List<RaceLeaderboardByCourseDto> Leaderboards { get; init; }
    public required LocationInfoWithRank LocationInfoWithRank { get; init; }
    public required string RaceKickOffDate { get; init; }
    public required string RaceName { get; init; }
    public required string RaceSeriesDescription { get; init; }
    public required RaceSeriesType RaceSeriesType { get; init; }
}
